doc/y_cable/design_doc.md

```diff
@@ -113,105 +143,119 @@ Vendors can have several implementations/ways to use this concept
 
-For example a typical module of the vendor can be like this
-```
+For example the base class would be like this
 
-	class Ycable(Ycablebase):
-            #all vendor modules inherit from Ycablebase
-	    def __init__(self):
+```python
+	class YCableBase(object):
+	    def __init__(self, port):
+		self.port = port
+                <function body here>
 
 	    def toggle_mux_to_torA(self, port):
-		#implement or raise exception if not implemented
-		raise NotImplementedError
+                <function body here>
+
 	    def toggle_mux_to_torB(self, port):
-		#implement or raise exception if not implemented
-		raise NotImplementedError
+                <function body here>
 
 	    def check_prbs(self, port):
-		#implement or raise exception if not implemented
-		raise NotImplementedError
+                <function body here>
 
 	    def only_credo(self, port):
-		raise NotImplementedError
+                <function body here>
 
 	    def only_amphenol(self, port):
-		raise NotImplementedError
+                <function body here>
 
-```
-the base class would be like this
 
 ```
-	class Ycablebase(object):
-	    def __init__(self, port):
-		self.port = port
+
+For example a typical module of the vendor can be like this
+```python
+
+	class YCable(YCableBase):
+            #All vendor modules inherit from YCableBase
+
+	    def __init__(self):
+                <function body here>
 
 	    def toggle_mux_to_torA(self, port):
-		raise NotImplementedError
+                <function body here>
+
 	    def toggle_mux_to_torB(self, port):
-		raise NotImplementedError
+                <function body here>
 
 	    def check_prbs(self, port):
-		raise NotImplementedError
+                <function body here>
 
 	    def only_credo(self, port):
-		raise NotImplementedError
+                <function body here>
 
 	    def only_amphenol(self, port):
-		raise NotImplementedError
+                <function body here>
 
 ```
-
 #### Implementation details
 - Now for xcvrd to use this solution, all it has to do is instantiate the class objects defined by importing the appropriate module. This should happen when xcvrd starts or if there is a change event  (xcvrd inserted/removed)
 
-```
-	from credo import y_cable
-	y_cable_port_Ethernet0 = Ycable(port)
+```python
+        # import the vendor to module mapping
+        from sonic_y_cable import y_cable_vendor_mapping
+ 
+        # This vendor/part_name determination subroutines will either read through eeprom
+        # or else read from the TRANSCEIVER_INFO_TABLE inside state db
+
+        vendor_name = get_vendor_name(port)
+
+        part_name = get_part_name(port)
 
+        module = y_cable_vendor_mapping.mapping[vendor_name][part_name]
 
+        from sonic_y_cable import vendor_name.part_name.module.YCable as YCable
```

Comment:
Since we will not require a 1:1 mapping from module to part number, we probably need to utilize `importlib` here to import the module from the path provided in the mapping. See [here](https://github.com/Azure/sonic-utilities/pull/1450/files#diff-2c73a525e9e3f2eb117764c56eddae85f332e75a9fac100ca70b5a32b5a37447R4-R13) for an example.

Reply:
fixed